Code package problems on armv7l

EDIT: this issue was originally for a problem with computing dependencies for the `code` package, but a fix has since been applied. There are some more issues described in this thread that have not been fixed, please look through the thread before dismissing it as not helpful. Most of these issues appear to be armv7l related, but they may still apply to you anyway.

## Description
i'm trying to install vs code.
when doing
```shell
crew install code
```
nothing happens for a long time (literally half an hour) and it finally fails:

Output:
```shell
~ [ crew install code                                                                                                                       ] 11:10 AM
code: Visual Studio Code is a source code editor developed by Microsoft for Windows, Linux and macOS.
https://code.visualstudio.com/
version 1.30.1-1
Traceback (most recent call last):
        12679: from /usr/local/bin/crew:988:in `<main>'
        12678: from /usr/local/bin/crew:919:in `install_command'
        12677: from /usr/local/bin/crew:919:in `each'
        12676: from /usr/local/bin/crew:923:in `block in install_command'
        12675: from /usr/local/bin/crew:612:in `resolve_dependencies_and_install'
        12674: from /usr/local/bin/crew:669:in `resolve_dependencies'
        12673: from /usr/local/bin/crew:663:in `expand_dependencies'
        12672: from /usr/local/bin/crew:657:in `push_dependencies'
         ... 12667 levels...
            4: from /usr/local/lib/ruby/2.5.0/find.rb:49:in `block (2 levels) in find'
            3: from /usr/local/bin/crew:148:in `block in search'
            2: from /usr/local/bin/crew:109:in `set_package'
            1: from /usr/local/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:39:in `require'
/usr/local/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:123:in `rescue in require': stack level too deep (SystemStackError)
```

## Versions and system information
chromebrew: `crew --version`
1.1.2
ChromeOS: `cat /etc/lsb-release`
CHROMEOS_AUSERVER=https://tools.google.com/service/update2
CHROMEOS_BOARD_APPID={2FC0F0CF-1A55-DF3F-73E6-517389444085}
CHROMEOS_CANARY_APPID={90F229CE-83E2-4FAF-8479-E368A34938B1}
CHROMEOS_DEVSERVER=
CHROMEOS_RELEASE_APPID={2FC0F0CF-1A55-DF3F-73E6-517389444085}
CHROMEOS_RELEASE_BOARD=nyan_big-signed-mp-v2keys
CHROMEOS_RELEASE_BRANCH_NUMBER=113
CHROMEOS_RELEASE_BUILDER_PATH=nyan_big-release/R71-11151.113.0
CHROMEOS_RELEASE_BUILD_NUMBER=11151
CHROMEOS_RELEASE_BUILD_TYPE=Official Build
CHROMEOS_RELEASE_CHROME_MILESTONE=71
CHROMEOS_RELEASE_DESCRIPTION=11151.113.0 (Official Build) stable-channel nyan_big 
CHROMEOS_RELEASE_KEYSET=mp-v2
CHROMEOS_RELEASE_NAME=Chrome OS
CHROMEOS_RELEASE_PATCH_NUMBER=0
CHROMEOS_RELEASE_TRACK=stable-channel
CHROMEOS_RELEASE_VERSION=11151.113.0
DEVICETYPE=CHROMEBOOK
GOOGLE_RELEASE=11151.113.0
Model: acer chromebook 13

Architecture: `uname -m`
armv7l
## Addtional information
i had tried to install it before but everything on my machine was massively outdated. it didn't fail that way before updating though.

i have read #3106 but have no idea how to use it (if that's even possible and/or advisable)

i have tried setting both `LD_LIBRARY_PATH` and `LIBRARY_PATH` but it doesn't seem to have taken. the only error before the stack trace is still
```shell
/usr/local/bin/ld: cannot find /usr/lib/libpthread_nonshared.a
collect2: error: ld returned 1 exit status
```
both variables are still set to `/usr/local/lib` after this.

quick update:
I was prodding around to see if i could dirty fix it by editing the makefile that was used, but it is repopulated everytime I do `yarn install`. since the makefile was in node-pty's folder (apparently the terminal emulator for vs code), I tried `npm install node-pty` to try to make it stop doing that.

during this, i found that either a) node-pty is being compiled on install or b)it compiles something on postinstall.

i'm gearing towards option a) since the last cd `make` did was in `node-modules/node-pty/build`.

this compilation is failing with the exact same message (and stack trace) as yarn install so I think installing is failing because it fails to build node-pty. keep in mind that i might very well be very wrong about this; for example: why is it even building anything? node-pty is a node module, it should be js and not compiled.

my version of the file had vs code 1.30.1 instead of 1.30.2 but i replaced it anyway and still no dice:
´´´shell
crew/packages [ crew install code                                                                                                     norec ] 12:27 AM
code: Visual Studio Code is a source code editor developed by Microsoft for Windows, Linux and macOS.
https://code.visualstudio.com/
version 1.30.1-1
No precompiled binary available for your platform, downloading source...
                                                            #-=O=-      #          #           #                                                      
Archive downloaded
Unpacking archive, this may take awhile...
gcc (GCC) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Building from source, this may take a while...
Rename all *.la files to *.la_tmp
npm WARN deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
npm WARN deprecated graceful-fs@3.0.11: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@1.2.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm notice created a lockfile as package-lock.json. You should commit this file.
+ gulp@3.9.1
added 255 packages from 164 contributors and audited 1187 packages in 23.115s
found 5 vulnerabilities (1 low, 4 high)
  run `npm audit fix` to fix them, or `npm audit` for details
yarn install v1.12.3
$ node build/npm/preinstall.js
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/4] Resolving packages...
[2/4] Fetching packages...
info vscode-fsevents@0.3.10: The platform "linux" is incompatible with this module.
info "vscode-fsevents@0.3.10" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.4: The platform "linux" is incompatible with this module.
info "fsevents@1.2.4" is an optional dependency and failed compatibility check. Excluding it from installation.
info windows-foreground-love@0.1.0: The platform "linux" is incompatible with this module.
info "windows-foreground-love@0.1.0" is an optional dependency and failed compatibility check. Excluding it from installation.
info windows-mutex@0.2.1: The platform "linux" is incompatible with this module.
info "windows-mutex@0.2.1" is an optional dependency and failed compatibility check. Excluding it from installation.
info windows-process-tree@0.2.3: The platform "linux" is incompatible with this module.
info "windows-process-tree@0.2.3" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > gulp-tsb@2.0.5" has incorrect peer dependency "typescript@^2.0.10".
[4/4] Building fresh packages...
[8/13] ⠁ vscode-nsfw
[2/13] ⠁ keytar
[7/13] ⠂ spdlog
[9/13] ⠂ vscode-ripgrep
error /usr/local/tmp/crew/1.30.1.tar.gz.dir/vscode-1.30.1/node_modules/node-pty: Command failed.
Exit code: 1
Command: node scripts/install.js
Arguments: 
Directory: /usr/local/tmp/crew/1.30.1.tar.gz.dir/vscode-1.30.1/node_modules/node-pty
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@8.15.0 | linux | arm
gyp info spawn /usr/local/bin/python2
gyp info spawn args [ '/usr/local/share/nodebrew/node/v8.15.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/tmp/crew/1.30.1.tar.gz.dir/vscode-1.30.1/node_modules/node-pty/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/share/nodebrew/node/v8.15.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/chronos/user/.node-gyp/iojs-2.0.12/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/chronos/user/.node-gyp/iojs-2.0.12',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/share/nodebrew/node/v8.15.0/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/home/chronos/user/.node-gyp/iojs-2.0.12/<(target_arch)/iojs.lib',
gyp info spawn args   '-Dmodule_root_dir=/usr/local/tmp/crew/1.30.1.tar.gz.dir/vscode-1.30.1/node_modules/node-pty',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/usr/local/tmp/crew/1.30.1.tar.gz.dir/vscode-1.30.1/node_modules/node-pty/build'
  CXX(target) Release/obj.target/pty/src/unix/pty.o
../src/unix/pty.cc: In function ‘void pty_after_waitpid(uv_async_t*)’:
../src/unix/pty.cc:489:69: warning: ‘v8::Local<v8::Value> Nan::Callback::Call(v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const’ is deprecated [-Wdeprecated-declarations]
   Nan::Callback(cb).Call(Nan::GetCurrentContext()->Global(), 2, argv);
                                                                     ^
In file included from ../src/unix/pty.cc:20:0:
../node_modules/nan/nan.h:1596:3: note: declared here
   Call(v8::Local<v8::Object> target
   ^~~~
  SOLINK_MODULE(target) Release/obj.target/pty.node
/usr/local/bin/ld: cannot find /usr/lib/libpthread_nonshared.a
collect2: error: ld returned 1 exit status
make: *** [pty.target.mk:128: Release/obj.target/pty.node] Error 1
make: Leaving directory '/usr/local/tmp/crew/1.30.1.tar.gz.dir/vscode-1.30.1/node_modules/node-pty/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/share/nodebrew/node/v8.15.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Linux 3.10.18
gyp ERR! command "/usr/local/share/nodebrew/node/v8.15.0/bin/node" "/usr/local/share/nodebrew/node/v8.15.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/tmp/crew/1.30.1.tar.gz.dir/vscode-1.30.1/node_modules/node-pty




^CTraceback (most recent call last):
        11: from /usr/local/bin/crew:963:in `<main>'
        10: from /usr/local/bin/crew:905:in `install_command'
         9: from /usr/local/bin/crew:905:in `each'
         8: from /usr/local/bin/crew:909:in `block in install_command'
         7: from /usr/local/bin/crew:609:in `resolve_dependencies_and_install'
         6: from /usr/local/bin/crew:704:in `install'
         5: from /usr/local/bin/crew:478:in `build_and_preconfigure'
         4: from /usr/local/bin/crew:478:in `chdir'
         3: from /usr/local/bin/crew:495:in `block in build_and_preconfigure'
         2: from /usr/local/lib/crew/packages/code.rb:42:in `build'
         1: from /usr/local/lib/crew/lib/package.rb:143:in `system'
/usr/local/lib/crew/lib/package.rb:143:in `system': Interrupt

crew/packages [                                                                                                                       norec ] 12:29 AM
´´´

still no dice:
´´´shell
crew/packages [ crew install code                                                                                                   norec * ] 12:35 AM
code: Visual Studio Code is a source code editor developed by Microsoft for Windows, Linux and macOS.
https://code.visualstudio.com/
version 1.30.2
No precompiled binary available for your platform, downloading source...
                                                      -=O=-                                  #          #         #        #                          
Archive downloaded
Unpacking archive, this may take awhile...
gcc (GCC) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Building from source, this may take a while...
Rename all *.la files to *.la_tmp
npm WARN deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
npm WARN deprecated graceful-fs@3.0.11: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@1.2.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm notice created a lockfile as package-lock.json. You should commit this file.
+ gulp@3.9.1
added 255 packages from 164 contributors and audited 1187 packages in 20.292s
found 5 vulnerabilities (1 low, 4 high)
  run `npm audit fix` to fix them, or `npm audit` for details
yarn install v1.12.3
$ node build/npm/preinstall.js
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/4] Resolving packages...
[2/4] Fetching packages...
info vscode-fsevents@0.3.10: The platform "linux" is incompatible with this module.
info "vscode-fsevents@0.3.10" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.4: The platform "linux" is incompatible with this module.
info "fsevents@1.2.4" is an optional dependency and failed compatibility check. Excluding it from installation.
info windows-foreground-love@0.1.0: The platform "linux" is incompatible with this module.
info "windows-foreground-love@0.1.0" is an optional dependency and failed compatibility check. Excluding it from installation.
info windows-mutex@0.2.1: The platform "linux" is incompatible with this module.
info "windows-mutex@0.2.1" is an optional dependency and failed compatibility check. Excluding it from installation.
info windows-process-tree@0.2.3: The platform "linux" is incompatible with this module.
info "windows-process-tree@0.2.3" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > gulp-tsb@2.0.5" has incorrect peer dependency "typescript@^2.0.10".
[4/4] Building fresh packages...
[8/13] ⠄ vscode-nsfw
[2/13] ⠄ keytar
[6/13] ⠄ node-pty
[9/13] ⠄ vscode-ripgrep
error /usr/local/tmp/crew/1.30.2.tar.gz.dir/vscode-1.30.2/node_modules/node-pty: Command failed.
Exit code: 1
Command: node scripts/install.js
Arguments: 
Directory: /usr/local/tmp/crew/1.30.2.tar.gz.dir/vscode-1.30.2/node_modules/node-pty
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@8.15.0 | linux | arm
gyp info spawn /usr/local/bin/python2
gyp info spawn args [ '/usr/local/share/nodebrew/node/v8.15.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/tmp/crew/1.30.2.tar.gz.dir/vscode-1.30.2/node_modules/node-pty/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/share/nodebrew/node/v8.15.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/chronos/user/.node-gyp/iojs-2.0.12/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/chronos/user/.node-gyp/iojs-2.0.12',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/share/nodebrew/node/v8.15.0/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/home/chronos/user/.node-gyp/iojs-2.0.12/<(target_arch)/iojs.lib',
gyp info spawn args   '-Dmodule_root_dir=/usr/local/tmp/crew/1.30.2.tar.gz.dir/vscode-1.30.2/node_modules/node-pty',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/usr/local/tmp/crew/1.30.2.tar.gz.dir/vscode-1.30.2/node_modules/node-pty/build'
  CXX(target) Release/obj.target/pty/src/unix/pty.o
../src/unix/pty.cc: In function ‘void pty_after_waitpid(uv_async_t*)’:
../src/unix/pty.cc:489:69: warning: ‘v8::Local<v8::Value> Nan::Callback::Call(v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const’ is deprecated [-Wdeprecated-declarations]
   Nan::Callback(cb).Call(Nan::GetCurrentContext()->Global(), 2, argv);
                                                                     ^
In file included from ../src/unix/pty.cc:20:0:
../node_modules/nan/nan.h:1596:3: note: declared here
   Call(v8::Local<v8::Object> target
   ^~~~
  SOLINK_MODULE(target) Release/obj.target/pty.node
/usr/local/bin/ld: cannot find /usr/lib/libpthread_nonshared.a
collect2: error: ld returned 1 exit status
make: *** [pty.target.mk:128: Release/obj.target/pty.node] Error 1
make: Leaving directory '/usr/local/tmp/crew/1.30.2.tar.gz.dir/vscode-1.30.2/node_modules/node-pty/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/share/nodebrew/node/v8.15.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Linux 3.10.18
gyp ERR! command "/usr/local/share/nodebrew/node/v8.15.0/bin/node" "/usr/local/share/nodebrew/node/v8.15.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/tmp/crew/1.30.2.tar.gz.dir/vscode-1.30.2/node_modules/node-pty




^CTraceback (most recent call last):
        11: from /usr/local/bin/crew:963:in `<main>'
        10: from /usr/local/bin/crew:905:in `install_command'
         9: from /usr/local/bin/crew:905:in `each'
         8: from /usr/local/bin/crew:909:in `block in install_command'
         7: from /usr/local/bin/crew:609:in `resolve_dependencies_and_install'
         6: from /usr/local/bin/crew:704:in `install'
         5: from /usr/local/bin/crew:478:in `build_and_preconfigure'
         4: from /usr/local/bin/crew:478:in `chdir'
         3: from /usr/local/bin/crew:495:in `block in build_and_preconfigure'
         2: from /usr/local/lib/crew/packages/code.rb:43:in `build'
         1: from /usr/local/lib/crew/lib/package.rb:143:in `system'
/usr/local/lib/crew/lib/package.rb:143:in `system': Interrupt
´´´

--- a/packages/code.rb
+++ b/packages/code.rb
@@ -13,7 +13,7 @@ class Code < Package
 
   description 'Visual Studio Code is a source code editor developed by Microsoft for Windows, Linux and macOS.'
   homepage 'https://code.visualstudio.com/'
-  version '1.30.2'
+  version '1.30.2-1'
   source_url 'https://github.com/Microsoft/vscode/archive/1.30.2.tar.gz'
   source_sha256 '7bf1c9c3c2814146906a6e38188d2f3284e128929e27c2c2dd0bcd7c6422be5e'
 
@@ -25,7 +25,6 @@ class Code < Package
   depends_on 'nodebrew'
   depends_on 'yarn' => :build
   depends_on 'gtk2'
-  depends_on 'xauth'
   depends_on 'ld_default'
   depends_on 'libsecret'
   depends_on 'libgconf'
@@ -54,7 +53,7 @@ class Code < Package
     FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/share"
     FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/bin"
     system 'mv', "../VSCode-linux-#{@arch}", "#{CREW_DEST_PREFIX}/share/code"
-    system "sed -i -e 's,ELECTRON_RUN_AS_NODE=1 ,,g' #{CREW_DEST_PREFIX}/share/code/bin/code-oss"
+    system "sed -i -e '/CLI=/d' -e 's,\"\\$CLI\" ,,g' -e 's,ELECTRON_RUN_AS_NODE=1 ,,g' #{CREW_DEST_PREFIX}/share/code/bin/code-oss"
     # ^^^ Do not remove this line.
     system 'ln', '-s', '../share/code/bin/code-oss', "#{CREW_DEST_PREFIX}/bin/code"
     system 'ln', '-s', '../share/code/bin/code-oss', "#{CREW_DEST_PREFIX}/bin/code-oss"
@@ -63,7 +62,7 @@ class Code < Package
   def self.postinstall
     puts
     puts 'Congratulations! You have installed Visual Studio Code on Chrome OS!'.lightgreen
-    puts 'Now, please run \'code-oss\' to start Visual Studio.'.lightgreen
+    puts 'Now, please run \'code-oss\' to start Code.'.lightgreen
     puts 'Happy coding!'.lightgreen
     puts
   end

